shared/src/clients/whatsapp-client.class/index.ts

```diff
@@ -59,4 +64,36 @@ export default class WhatsappClient {
     }
     return res
   }
+
+  public async getTemplates(wabaId: string): Promise<WhatsappTemplate[]> {
+    const res = await this.request({
+      method: 'get',
+      url: `${wabaId}/message_templates`,
+    }).catch((e) => {
+      // this would mean whatsapp server is down
+      throw new Error(e)
+    })
+    // even if whatsapp returned an error, it would still have status code: 200
+    // we must check for the error object instead
+    if (res.error) {
+      throw new Error(res)
```

Comment:
Can the message be more specific than the stringified `res`? Like `Error(res.message)` or something (this is just a shot in the dark as I'm not aware of the structure of this obj